### PR TITLE
Add toggle entry verification button

### PR DESCRIPTION
**Pull request by frozenhelium**
Depends on https://github.com/the-deep/server/pull/684

### CODE REVIEW

File: src/hooks/useProjectMemberListQuery.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        pending,
+        response,
+    ] = useRequest<MultiResponse<Member>>({
+        url: `server://v2/projects/${projectId}/members/`,
+        method: 'GET',
+        query: memberFieldQuery,
+        autoTrigger: true,
+        onFailure: (_, errorBody) => {
+            notifyOnFailure(_ts('entryReview', 'reviewHeading'))({ error: errorBody });
+        },
```
Comment (AdityaKhatri) [inline src/hooks/useProjectMemberListQuery.ts]: use notifyError you created

File: src/components/general/EntryCommentFormForModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    } = props;
+
+    const [faramValues, setFaramValues] = React.useState<T | {}>(initialValue);
+    const [faramErrors, setFaramErrors] = React.useState<FaramErrors>({});
+    const [
+        projectMembersPending,
+        projectMembersResponse,
+    ] = useProjectMemberListQuery(projectId);
+
+    const handleFaramChange = React.useCallback((newFaramValues, newFaramErrors) => {
```
Comment (tnagorra) [inline src/components/general/EntryCommentFormForModal/index.tsx]: Include typings

File: src/components/general/EntryCommentFormForModal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        setFaramValues(newFaramValues);
+        setFaramErrors(newFaramErrors);
+    }, []);
+
+    const handleFaramValidationSuccess = React.useCallback((_, newFaramValues) => {
```
Comment (tnagorra) [inline src/components/general/EntryCommentFormForModal/index.tsx]: Include typings

File: src/components/general/EntryCommentFormForModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+function EntryCommentFormForModal<T extends {}>(props: EntryCommentFormForModalProps<T>) {
+    const {
+        initialValue = {},
+        disabled,
+        onValidationSuccess,
+        projectId,
+        className,
+    } = props;
+
+    const [faramValues, setFaramValues] = React.useState<T | {}>(initialValue);
```
Comment (tnagorra) [inline src/components/general/EntryCommentFormForModal/index.tsx]: Why create a union of T and {}

File: src/components/general/ToggleEntryVerification/index.tsx [hunk 5]
```diff
@@ -28,17 +23,22 @@ import styles from './styles.scss';
...
+    projectId: DatabaseEntityBase['id'];
     onChange: (newValue: boolean) => void;
     onPendingStatusChange?: (newValue: boolean) => void;
     value: boolean;
     tooltip?: string;
     disabled?: boolean;
 }
 
+// Verify
+const verifyFormData = { commentType: 3 };
```
Comment (tnagorra) [inline src/components/general/ToggleEntryVerification/index.tsx]:
```suggestion
const COMMENT_TYPE_VERIFY = 3;
const verifyFormData = { commentType: COMMENT_TYPE_VERIFY };
```

File: src/components/general/ToggleEntryVerification/index.tsx [hunk 6]
```diff
@@ -52,19 +52,15 @@ function ToggleEntryVerification(props: ToggleEntryVerificationProps) {
         setCommentModalHidden,
     ] = useModalState(false);
 
-    const [comment, setComment] = useInputValue('');
+    const [unverifyFormData, setUnverifyFormData] = React.useState({
+        // Unverify
+        commentType: 4,
```
Comment (tnagorra) [inline src/components/general/ToggleEntryVerification/index.tsx]: Please use constants

File: src/hooks/stateManagement.ts [hunk 7]
```diff
@@ -24,14 +24,18 @@ export function useModalState(initialValue: boolean): [
     return [visible, setVisible, setHidden, setVisibility];
 }
 
-export function useInputValue(initialValue: string | undefined): [
-    string | undefined,
-    (v: string | undefined, n: string | undefined, e: React.FormEvent<HTMLTextAreaElement>) => void
+export function useInputValue<T>(initialValue: T[] | string | undefined): [
```
Comment (tnagorra) [inline src/hooks/stateManagement.ts]:
```suggestion
export function useInputValue<T>(initialValue: T | undefined): [
```

File: src/hooks/stateManagement.ts [hunk 8]
```diff
@@ -24,14 +24,18 @@ export function useModalState(initialValue: boolean): [
...
+    T[] | string | undefined,
+    (
+        v: T[] | string | undefined,
+        n: string | undefined,
+        e: React.FormEvent<HTMLTextAreaElement>,
```
Comment (tnagorra) [inline src/hooks/stateManagement.ts]: This is dangerous.

File: src/components/general/ToggleEntryApproval/index.tsx [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+    projectId: DatabaseEntityBase['id'];
+    onChange: (newValue: boolean, newCount: number) => void;
+    onPendingStatusChange?: (newValue: boolean) => void;
+    value: boolean;
+    tooltip?: string;
+    disabled?: boolean;
+    approvalCount: number;
+}
+
+const approveFormData = { commentType: 1 };
```
Comment (tnagorra) [inline src/components/general/ToggleEntryApproval/index.tsx]: Please use constants

File: src/components/general/ToggleEntryApproval/index.tsx [hunk 10]
```diff
@@ -0,0 +1,181 @@
...
+    } = props;
+
+    const [
+        commentModalShown,
+        setCommentModalVisible,
+        setCommentModalHidden,
+    ] = useModalState(false);
+
+    const [unapproveFormData, setUnapproveFormData] = React.useState({
+        commentType: 2,
```
Comment (tnagorra) [inline src/components/general/ToggleEntryApproval/index.tsx]: Please use constants

File: src/hooks/stateManagement.ts [hunk 11]
```diff
@@ -24,14 +24,18 @@ export function useModalState(initialValue: boolean): [
     return [visible, setVisible, setHidden, setVisibility];
 }
 
-export function useInputValue(initialValue: string | undefined): [
-    string | undefined,
-    (v: string | undefined, n: string | undefined, e: React.FormEvent<HTMLTextAreaElement>) => void
+export function useInputValue<T>(initialValue: T[] | T | undefined): [
```
Comment (tnagorra) [inline src/hooks/stateManagement.ts]:
```suggestion
export function useInputValue<T>(initialValue: T | undefined): [
```

**Pull request merged**